modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/RaftClientService.java

```diff
@@ -103,4 +103,16 @@ Future<Message> timeoutNow(final PeerId peerId, final RpcRequests.TimeoutNowRequ
      */
     Future<Message> readIndex(final PeerId peerId, final RpcRequests.ReadIndexRequest request, final int timeoutMs,
         final RpcResponseClosure<RpcRequests.ReadIndexResponse> done);
+
+    /**
+     * Send a get-leader-and-term request and handle the response with done.
+     *
+     * @param peerId destination peer ID
+     * @param request request data
+     * @param timeoutMs timeout millis
+     * @param done callback
+     * @return a future result
+     */
+    Future<Message> getLeaderAndTerm(final PeerId peerId, final CliRequests.GetLeaderRequest request, final int timeoutMs,
```

Comment:
Should we instead send a redirect-proposal to client? I mean that instead of sending GetLeaderRequest to leader if original client-based one was processed on follower `case STATE_FOLLOWER:` we way response with redirect proposal in order to force client to re-send GetLeaderRequest to proposed leader. I though that it's common way to solve the leader-miss issue.